misc/vtk/src/processors/vtktovolume.cpp

```diff
@@ -0,0 +1,422 @@
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *********************************************************************************/
+
+#include <inviwo/vtk/processors/vtktovolume.h>
+#include <inviwo/core/util/filesystem.h>
+#include <inviwo/core/datastructures/volume/volume.h>
+#include <inviwo/core/datastructures/volume/volumeramprecision.h>
+#include <inviwo/core/common/inviwoapplication.h>
+#include <inviwo/core/network/networklock.h>
+#include <inviwo/core/util/clock.h>
+#include <inviwo/core/util/formats.h>
+#include <modules/base/algorithm/dataminmax.h>
+
+#include <warn/push>
+#include <warn/ignore/all>
+#include <vtkCallbackCommand.h>
+#include <warn/pop>
+
+#include <vtkInformation.h>
+#include <vtkDataSet.h>
+#include <vtkCellData.h>
+#include <vtkPointData.h>
+#include <vtkDataArray.h>
+#include <vtkRectilinearGrid.h>
+#include <vtkStructuredGrid.h>
+#include <vtkImageData.h>
+#include <vtkMatrix4x4.h>
+#include <vtkType.h>
+
+#include <fstream>
+#include <inviwo/vtk/util/vtkutil.h>
+#include <fmt/format.h>
+
+namespace inviwo {
+
+// The Class Identifier has to be globally unique. Use a reverse DNS naming scheme
+const ProcessorInfo VTKtoVolume::processorInfo_{
+    "org.inviwo.VTKtoVolume",     // Class identifier
+    "VTK to Volume",              // Display name
+    "VTK",                        // Category
+    CodeState::Experimental,      // Code state
+    "CPU, Data Conversion, VTK",  // Tags
+};
+const ProcessorInfo VTKtoVolume::getProcessorInfo() const { return processorInfo_; }
+
+VTKtoVolume::VTKtoVolume()
+    : Processor()
+    , outport_("VolumeOutport")
+    , inport_("VTKDataObjectInport")
+    , useCellData_("useCellData", "Use cell data", false)
+    , dataArrays_("availableArrays", "Data Arrays")
+    , dataFormats_("existingDataFormats", "Data formats")
+    , convertButton_("triggerConvert", "Convert")
+    , dataRange_("dataRange", "Data range", vec2(0, 1))
+    , reactToChange_(true)
+    , convertData_(false)
+    , formerArraySelection_() {
+
+    addPort(outport_);
+    addPort(inport_);
+    addProperty(useCellData_);
+    addProperty(dataFormats_);
+    addProperty(dataArrays_);
+    addProperty(convertButton_);
+    addProperty(dataRange_);
+
+    inport_.onChange([this]() {
+        if (reactToChange_) updateFormats();
+    });
+    useCellData_.onChange([this]() { updateFormats(); });
+    dataFormats_.onChange([this]() {
+        if (!reactToChange_) return;
+        dataArrays_.clear();
+        updateArrays();
+    });
+    convertButton_.onChange([this]() { convertData_ = true; });
+    dataArrays_.onChange([this]() {
+        if (reactToChange_) updateAvailableArrays();
+    });
+}
+
+void VTKtoVolume::process() {
+    if (convertData_) {
+        convertData();
+        convertData_ = false;
```

Comment:
I assume this is a dirty flag? Rename to dataIsDirty_ or something.

You can also override the isReady function instead.

Reply:
`isReady` is not virtual, so I won't do that one.

Reply:
One can change `isReady`-behaviour with overloading: https://github.com/inviwo/inviwo/blob/2205ee567b625a65c8aeccb78cd56ea15d5c4e6c/modules/base/src/processors/imagesource.cpp#L71-L74